test/DebugInfo/while.swift

```diff
@@ -9,5 +9,5 @@ while let val = yieldValues() {
 }
 while let val = yieldValues() {
 // CHECK: !DILocalVariable(name: "val", scope: ![[SCOPE2:[0-9]+]]
-// CHECK: ![[SCOPE2]] = distinct !DILexicalBlock(scope: ![[MAIN]]
+// CHECK: ![[SCOPE2]] = distinct !DILexicalBlock(scope: ![[MAIN2:[0-9]+]]
```

Comment:
I find this change a little scary. What is happening here? Shouldn't those two lexical blocks be in the same scope?

Reply:
visitTopLevelDecl is called every time you visit a new function so if you have

```
func blah() {
}
func foo() {
}
```

they will end up in two different scopes. It might an acceptable bloat, what do you think?

Reply:
they're both top level decls.